examples/inference-deployments/mpt/mpt_7b_ft_handler.py

```diff
@@ -0,0 +1,248 @@
+            int8_mode=int8_mode,
+            weights_data_type=weights_data_type,
+            layernorm_eps=layernorm_eps,
+            use_attention_linear_bias=use_attention_linear_bias,
+            has_positional_encoding=has_positional_encoding,
+            shared_contexts_ratio=shared_contexts_ratio)
+        if not self.model.load(ckpt_path=model_path):
+            raise RuntimeError(
+                'Could not load model from a FasterTransformer checkpoint')
+
+        self.tokenizer = AutoTokenizer.from_pretrained(self.model_name,
+                                                       trust_remote_code=True)
+
+    def _parse_inputs(self, inputs: Dict[str, Any]):
+        if self.INPUT_STRINGS_KEY not in inputs or not isinstance(
+                inputs[self.INPUT_STRINGS_KEY], list):
+            raise RuntimeError(
+                'Input strings must be provided as a list to generate call')
+
+        generate_input = inputs[self.INPUT_STRINGS_KEY]
+        batch_size = len(generate_input)
+
+        # Set default generate kwargs
+        generate_kwargs = copy.deepcopy(self.DEFAULT_GENERATE_KWARGS)
+
+        # If request contains any additional kwargs, add them to generate_kwargs
+        for k, v in inputs.items():
+            if k not in [self.INPUT_STRINGS_KEY]:
+                generate_kwargs[k] = v
+        generate_kwargs['top_k'] *= torch.ones(batch_size, dtype=torch.int32)
+        generate_kwargs['top_p'] *= torch.ones(batch_size, dtype=torch.float32)
+        generate_kwargs['temperature'] *= torch.ones(batch_size,
+                                                     dtype=torch.float32)
+        repetition_penalty = generate_kwargs['repetition_penalty']
+        generate_kwargs[
+            'repetition_penalty'] = None if repetition_penalty == 1.0 else repetition_penalty * torch.ones(
+                batch_size, dtype=torch.float32)
+        presence_penalty = generate_kwargs['presence_penalty']
+        generate_kwargs[
+            'presence_penalty'] = None if presence_penalty == 0.0 else presence_penalty * torch.ones(
+                batch_size, dtype=torch.float32)
+        generate_kwargs['beam_search_diversity_rate'] *= torch.ones(
+            batch_size, dtype=torch.float32)
+        generate_kwargs['len_penalty'] *= torch.ones(size=[batch_size],
+                                                     dtype=torch.float32)
+        generate_kwargs['min_length'] *= torch.ones(size=[batch_size],
+                                                    dtype=torch.int32)
+        if generate_kwargs['random_seed']:
+            generate_kwargs['random_seed'] = torch.randint(0,
+                                                           10000,
+                                                           size=[batch_size],
+                                                           dtype=torch.int64)
+
+        return generate_input, generate_kwargs
+
+    def predict(self, **inputs: Dict[str, Any]):
```

Comment:
This will have to change to work with batching. Inputs will be of type List[Dict[str,Any]]. But that can be done in a future pr.